_docs/rfc/scaffolding_new_modules.md

```diff
@@ -0,0 +1,402 @@
+to check each time. Both of these are unreasonable to expect for sufficiently large infrastructure deployments with many
+possible modules and dependencies.
+
+The goal of this RFC is to propose an approach to handling these basic tasks that minimizes the room for operator
+errors.
+
+
+## Proposed solution
+
+We will introduce a new command `terragrunt new`, which provides a code generation interface for solving each of the
+scenarios described above. The goal of `terragrunt new` is to provide the ability to templatize your configurations with
+annotations in the code that reminds the user what pieces need to change for each scenario.
+
+It is important that the templates are customizable. Not all users have the same requirements, so allowing users to
+customize the templates to their needs is important. The goal is to allow users to define their own standards as
+templates that then is used by their engineers to generate new configurations that conform to their standards, which may
+be different from Gruntwork standards. I envision Gruntwork providing a set of base templates that can be used by
+default, which can be overridden by either forking our `infrastructure-templates` repository or by building new ones
+from scratch.
+
+`terragrunt new` should support the following scenarios:
+
+- `terragrunt new component`, which will add a new component to an existing environment. The inputs for this template
+  should be:
+    - Environment: which live environment folder should the component be added to?
+    - Component module: which module should be used for the component?
+
+- `terragrunt new environment`, which will add a new environment to an existing region. The inputs for this template
+  should be:
+    - Location: which live region/account folder should the environment be added to?
+    - Stack: which stack should be used? This assumes you have different kinds of stacks in your environment, where a
+      stack is a set of components.
+
+- `terragrunt new region`, which will add a new region to an existing account. The inputs for this template should be:
+    - Account: which live account folder should the region be added to? (NOTE: `account` may not be the right
+      terminology here, as it does not mesh with GCP where you have Projects).
+
+- `terragrunt new account`, which will add a new account. The inputs for this template should be:
+    - Remote state info: the minimum information necessary to setup the remote state config.
+
+- `terragrunt new wrapper-module`, which will add a new wrapper module (what you have in `infrastructure-modules`). Note
+  that as such, this will be rendered into `infrastructure-modules`, as opposed to `infrastructure-live` like the other
+  sub commands. The inputs for this template should be:
+    - Module name: what to call this module.
+    - Library Module source: what the underlying library module you are basing the wrapper on.
```

Comment:
Oh, cool idea. If you can use this with modules in the Terraform Registry and Gruntwork IaC Lib, that would be very valuable.

Reply:
Consider generating a scaffold for Terratest tests too.